ext/ManoptJuMPExt.jl

```diff
@@ -304,36 +327,11 @@ function MOI.optimize!(model::Optimizer)
             model.variable_primal_start[i]
         end for i in eachindex(model.variable_primal_start)
     ]
-    backend = MOI.Nonlinear.SparseReverseMode()
-    vars = [MOI.VariableIndex(i) for i in eachindex(model.variable_primal_start)]
-    evaluator = MOI.Nonlinear.Evaluator(model.nlp_model, backend, vars)
-    MOI.initialize(evaluator, [:Grad])
-    function eval_f_cb(M, x)
-        if model.sense == MOI.FEASIBILITY_SENSE
-            return 0.0
-        end
-        obj = MOI.eval_objective(evaluator, JuMP.vectorize(x, _shape(model.manifold)))
-        if model.sense == MOI.MAX_SENSE
-            obj = -obj
-        end
-        return obj
-    end
-    function eval_grad_f_cb(M, X)
-        x = JuMP.vectorize(X, _shape(model.manifold))
-        grad_f = zeros(length(x))
-        if model.sense == MOI.FEASIBILITY_SENSE
-            grad_f .= zero(eltype(grad_f))
-        else
-            MOI.eval_objective_gradient(evaluator, grad_f, x)
-        end
-        if model.sense == MOI.MAX_SENSE
-            LinearAlgebra.rmul!(grad_f, -1)
-        end
-        reshaped_grad_f = JuMP.reshape_vector(grad_f, _shape(model.manifold))
-        return ManifoldDiff.riemannian_gradient(model.manifold, X, reshaped_grad_f)
+    objective = model.objective
+    if model.sense == MOI.MAX_SENSE
+        objective = -objective
```

Comment:
This is the main difficulty I see, how do we multiply an `AbstractManifoldObjective` by `-1` ?

Reply:
Yes, also we would need to support all types of different `AbstractManifolObjective`. Maybe we can start with just throwing an error and forcing the user to use a minimization ?

Reply:
One idea could be to write a new wrapper like https://manoptjl.org/stable/plans/objective/#Manopt.EmbeddedManifoldObjective
That one internally stores the objective defined on the embedding space and converts every call to the cost/grad/... to a Riemannian call.

We could do the same with a generic “ScaledManifoldObjective” – and even define the unary minus operator for abstracts then. I would maybe not do that for the constraint objectives but at least for all up to hessian?

Reply:
That would be a solution, maybe you can do a separate PR with this feature and then we continue this one once this is merged ?

Reply:
Yes, sure.